src/IR/LambdaLift.hs

```diff
@@ -209,20 +239,31 @@ liftLambdas (I.Match s arms t) = do
 liftLambdas lit@I.Lit{}  = return lit
 liftLambdas dat@I.Data{} = return dat
 
+-- | Entry point for traversing let bindings.
+liftLambdasInLet :: (I.Binder, I.Expr Poly.Type) -> LiftFn (I.Expr Poly.Type)
+liftLambdasInLet (b, expr) = do
+  let bindingName = maybe "wildcard" ident b
+  (liftedLetBody, bodyFrees) <- descend (Just $ fromString bindingName)
+    $ liftLambdas expr
+  modify $ \st -> st { freeTypes = bodyFrees }
+  return liftedLetBody
+
 -- | Entry point for traversing the arms of match expressions.
 liftLambdasInArm
   :: (I.Alt, I.Expr Poly.Type) -> LiftFn (I.Alt, I.Expr Poly.Type)
 liftLambdasInArm (I.AltLit l, arm) = do
-  (liftedArm, armFrees) <- descend $ liftLambdas arm
+  (liftedArm, armFrees) <- descend Nothing $ liftLambdas arm
   modify $ \st -> st { freeTypes = armFrees }
   return (I.AltLit l, liftedArm)
 liftLambdasInArm (I.AltDefault b, arm) = do
-  (liftedArm, armFrees) <-
-    descend $ F.forM_ b addCurrentScope >> liftLambdas arm
+  (liftedArm, armFrees) <- descend Nothing $ do
+    F.forM_ b addCurrentScope
```

Comment:
nit: `import Control.Monad (formM_)` instead of using a qualified import, for consistency with other modules